web/src/ShareSnapshotModal.tsx

```diff
@@ -1,22 +1,50 @@
+import { CameraAlt } from "@material-ui/icons"
+import { saveAs } from "file-saver"
 import cookies from "js-cookie"
 import React, { PureComponent } from "react"
 import Modal from "react-modal"
+import styled from "styled-components"
 import intro from "./assets/png/share-snapshot-intro.png"
 import { ReactComponent as ArrowSvg } from "./assets/svg/arrow.svg"
+import { linkToTiltDocs, TiltDocsPage } from "./constants"
+import { Flag, useFeatures } from "./feature"
+import FloatDialog from "./FloatDialog"
+import { InstrumentedButton } from "./instrumentedComponents"
 import "./ShareSnapshotModal.scss"
+import { Color, Font, FontSize, SizeUnit } from "./style-helpers"
 
-type props = {
-  handleSendSnapshot: () => void
+type ShareSnapshotModalProps = {
+  handleSendSnapshot: (s: Proto.webviewSnapshot) => void
+  getSnapshot: () => Proto.webviewSnapshot
   handleClose: () => void
   snapshotUrl: string
   tiltCloudUsername: string | null
   tiltCloudSchemeHost: string
   tiltCloudTeamID: string | null
   isOpen: boolean
   highlightedLines: number | null
+  dialogAnchor: HTMLElement | null
 }
 
-export default class ShareSnapshotModal extends PureComponent<props> {
+// TODO (lizz): When offline snapshots are the default method for snapshots
+// generation, cloud snapshot code can be refactored
+export default function ShareSnapshotModal(props: ShareSnapshotModalProps) {
```

Comment:
Pure bikeshed, but `{Upload,Download}SnapshotModal` are kinda confusing in this context, maybe `LocalSnapshotModal` & `CloudSnapshotModal`?

Reply:
ha, yeah. I named DownloadSnapshotModal without thinking and thought it was cutely chaotic (in that the names make sense in isolation but pretty misleading in combination) but had meant to change them before merging.

Reply:
done!